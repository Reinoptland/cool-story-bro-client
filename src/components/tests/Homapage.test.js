import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";

import Homepage from "../Homepage";

describe("<Homepage >", () => {
  const props = {
    title: "My homepage!",
    color: "#435234",
    backgroundColor: "#435234",
    id: 1,
    description: "some description",
  };

  test("should render without props", () => {
    const component = renderer.create(<Homepage />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("should render with all props", () => {
    const component = renderer.create(<Homepage {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("should render link if showLink prop is true", () => {
    const component = renderer.create(
      <MemoryRouter>
        <Homepage {...props} showLink={true} />
      </MemoryRouter>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
