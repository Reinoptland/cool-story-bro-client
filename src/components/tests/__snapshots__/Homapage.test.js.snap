// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Homepage > should render link if showLink prop is true 1`] = `
<div
  className="jumbotron"
  style={
    Object {
      "backgroundColor": "#435234",
      "color": "#435234",
    }
  }
>
  <h1>
    My homepage!
  </h1>
  <p>
    some description
  </p>
  <a
    href="/homepages/1"
    onClick={[Function]}
  >
    <button
      className="btn btn-primary"
      disabled={false}
      type="button"
    >
      Visit page
    </button>
  </a>
</div>
`;

exports[`<Homepage > should render with all props 1`] = `
<div
  className="jumbotron"
  style={
    Object {
      "backgroundColor": "#435234",
      "color": "#435234",
    }
  }
>
  <h1>
    My homepage!
  </h1>
  <p>
    some description
  </p>
</div>
`;

exports[`<Homepage > should render without props 1`] = `
<div
  className="jumbotron"
  style={
    Object {
      "backgroundColor": undefined,
      "color": undefined,
    }
  }
>
  <h1 />
  <p />
</div>
`;
